src/pages/iou/request/step/IOURequestStepDistance.js

```diff
@@ -0,0 +1,233 @@
+    const hasRoute = TransactionUtils.hasRoute(transaction);
+    const validatedWaypoints = TransactionUtils.getValidWaypoints(waypoints);
+    const previousValidatedWaypoints = usePrevious(validatedWaypoints);
+    const haveValidatedWaypointsChanged = !_.isEqual(previousValidatedWaypoints, validatedWaypoints);
+    const isRouteAbsentWithoutErrors = !hasRoute && !hasRouteError;
+    const shouldFetchRoute = (isRouteAbsentWithoutErrors || haveValidatedWaypointsChanged) && !isLoadingRoute && _.size(validatedWaypoints) > 1;
+
+    useEffect(() => {
+        MapboxToken.init();
+        return MapboxToken.stop;
+    }, []);
+
+    useEffect(() => {
+        if (isOffline || !shouldFetchRoute) {
+            return;
+        }
+
+        Transaction.getRouteForDraft(transactionID, validatedWaypoints);
+    }, [shouldFetchRoute, transactionID, validatedWaypoints, isOffline]);
+
+    useEffect(() => {
+        if (numberOfWaypoints <= numberOfPreviousWaypoints) {
+            return;
+        }
+        scrollViewRef.current.scrollToEnd({animated: true});
+    }, [numberOfPreviousWaypoints, numberOfWaypoints]);
+
+    const navigateBack = () => {
+        Navigation.goBack(backTo || ROUTES.HOME);
+    };
+
+    /**
+     * Takes the user to the page for editing a specific waypoint
+     * @param {Number} index of the waypoint to edit
+     */
+    const navigateToWaypointEditPage = (index) => {
+        Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_WAYPOINT.getRoute(iouType, transactionID, reportID, index));
+    };
+
+    const navigateToNextStep = useCallback(() => {
+        if (backTo) {
+            Navigation.goBack(backTo);
+            return;
+        }
+
+        // If a reportID exists in the report object, it's because the user started this flow from using the + button in the composer
+        // inside a report. In this case, the participants can be automatically assigned from the report and the user can skip the participants step and go straight
+        // to the confirm step.
+        if (report.reportID) {
+            IOU.setMoneyRequestParticipantsFromReport(transactionID, report);
+            Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_CONFIRMATION.getRoute(iouType, transactionID, reportID));
+            return;
+        }
+
+        // If there was no reportID, then that means the user started this flow from the global + menu
+        // and an optimistic reportID was generated. In that case, the next step is to select the participants for this request.
+        Navigation.navigate(ROUTES.MONEY_REQUEST_STEP_PARTICIPANTS.getRoute(iouType, transactionID, reportID));
+    }, [report, iouType, reportID, transactionID, backTo]);
+
+    const getError = () => {
+        // Get route error if available else show the invalid number of waypoints error.
+        if (hasRouteError) {
+            return ErrorUtils.getLatestErrorField(transaction, 'route');
+        }
+
+        if (_.size(validatedWaypoints) < 2) {
```

Comment:
Coming from https://github.com/Expensify/App/issues/29895#issuecomment-1792954251.

We missed a case when consecutive duplicate waypoints are present for more than 2 waypoints.